### PR TITLE
docs - update saml & sso

### PR DESCRIPTION
**Pull request by nllho**
A round of updates based on customer questions and other knowledge stored outside of our docs.
- [Sources for SSO updates](https://www.notion.so/metabase/Doc-updates-72e5f66f807e4dabbee43220e3ded6e6)
- [Sources for SAML updates](https://www.notion.so/metabase/Doc-updates-412c459d19e84e7c89d74020f79a007e)

Docs landing page 
- Add links to SAML guides under SSO

Admin guide landing page
- Add links to SAML guides under SSO

SSO landing page 
- Add routing to glossary, docs, and troubleshooting
- Add "checking if SSO works" section

SAML landing page 
- Add overview section
- Add routing to new Keycloak and Google SAML pages

Auth with SSO
- Add table of contents
- Add new sections:
  - Syncing user attributes at login
  - Changing a user's login method
  - Checking if SSO is working correctly
- Remove advanced LDAP section (info relocated to other sections)

New pages:
- Google SAML
- Keycloak SAML

### CODE REVIEW

File: docs/enterprise-guide/saml-google.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Setting up SAML with Google
```
Comment (nllho) [inline docs/enterprise-guide/saml-google.md:1]: @flamber I took this from your notes here https://github.com/metabase/metabase/issues/21290#issuecomment-1080997005 and made some slight updates based on the Google docs. 

Let me know if it's correct and if there's anything else we should add!

File: docs/enterprise-guide/sso-keycloak.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Enabling SSO with Keycloak
```
Comment (nllho) [inline docs/enterprise-guide/sso-keycloak.md:1]: @paoliniluis I took this from the instructions you wrote up. Let me know if it's correct, and whether it's still clear without the screenshots!

File: docs/administration-guide/10-single-sign-on.md [hunk 3]
```diff
@@ -82,20 +97,44 @@ As you can see below, if you have an **Accounting** group in both your LDAP serv
...
 - [LDAP troubleshooting guide](../troubleshooting-guide/ldap.md)
 - [Permissions overview](05-setting-permissions.md)
+
+## Syncing user attributes at login
+
+{% include plans-blockquote.html feature="Advanced authentication features" %}
+
+### Syncing attributes with LDAP
+
+LDAP user attributes are managed from your LDAP directory. If you've set up [data-sandboxing-docs][data-sandboxing-docs], these attributes will be [automatically passed to Metabase][getting-user-attributes].
```
Comment (jeff-bruemmer) [inline docs/administration-guide/10-single-sign-on.md:118]: Can you rephrase to active voice so it's clear what's managing and passing attributes?

File: docs/administration-guide/10-single-sign-on.md [hunk 4]
```diff
@@ -82,20 +97,44 @@ As you can see below, if you have an **Accounting** group in both your LDAP serv
...
+
+### Syncing attributes with Google
+
+User attributes can't be synced with regular Google Sign-In. You'll need to set up [Google SAML][google-saml-docs] or [JWT][jwt-docs] instead.
+
+## Changing a user's login method from email to SSO
```
Comment (jeff-bruemmer) [inline docs/administration-guide/10-single-sign-on.md:124]:
```suggestion
## Changing an account's login method from email to SSO
```

Unless it's awkward or incorrect to omit "user", avoid the term "user". Prefer people or account.

File: docs/administration-guide/10-single-sign-on.md [hunk 5]
```diff
@@ -82,20 +97,44 @@ As you can see below, if you have an **Accounting** group in both your LDAP serv
...
+
+You cannot use the Admin panel to update the authentication method for a user once they have set it up for the first time.
```
Comment (jeff-bruemmer) [inline docs/administration-guide/10-single-sign-on.md:126]:
```suggestion
Once a person creates an account, you cannot change the authentication method for that account.
```

Should we suggest the workaround here?

File: docs/administration-guide/10-single-sign-on.md [hunk 6]
```diff
@@ -82,20 +97,44 @@ As you can see below, if you have an **Accounting** group in both your LDAP serv
...
+
+Metabase accounts created with Google Sign-In do not have passwords and must use Google to sign in to Metabase.
+
+For Metabase accounts created with emails and passwords, you can:
+- Deactivate password authentication for all users from **Admin settings** > **Authentication**. You'll need to ask users to sign in with Google if they haven't already.
+- Manually update the user's login method in the Metabase application database. This option is not recommended unless you are very familiar with making changes to the application database.
+
+Note that you must have at least one user with email and password login. This ensures that you won't get locked out of Metabase if there are any problems with your SSO provider.
```
Comment (jeff-bruemmer) [inline docs/administration-guide/10-single-sign-on.md:134]:
```suggestion
For Metabase accounts created with emails and passwords, you can:
- Deactivate password authentication for all users from **Admin settings** > **Authentication**. You'll need to ask people to sign in with Google (if they haven't already).
- Manually update the account's login method in the Metabase application database. This option is not recommended unless you're  familiar with making changes to the application database.

Note that you must have at least one account with email and password login. This account safeguards you from getting locked out of your Metabase if there are any problems with your SSO provider.
```

File: docs/enterprise-guide/authenticating-with-saml.md [hunk 7]
```diff
@@ -130,14 +130,18 @@ Your IdP may label it a little differently. Here are some of the names we've fou
 | OneLogin | Issuer URL                           |
 
 
-**SAML Identity Provider Certificate:** This is an encoded certificate that Metabase will use when connecting to the IdP URI. This will look like a big blob of text that you'll want to copy and paste carefully — the spacing is important! Your IdP might have you download this certificate as a file, which you'll then need to open up in a text editor in order to copy the contents to then paste into the box in Metabase. Again, different providers may have slightly different labels for this:
+**SAML Identity Provider Certificate:** This is an encoded certificate that Metabase will use when connecting to the IdP URI. This will look like a big blob of text that you'll want to copy and paste carefully — the spacing is important!
```
Comment (jeff-bruemmer) [inline docs/enterprise-guide/authenticating-with-saml.md:133]:
```suggestion
**SAML Identity Provider Certificate:** This is an encoded certificate that Metabase will use when connecting to the IdP URI. The certificate will look like a big blob of text that you'll want to copy and paste carefully — the spacing is important! 
```

File: docs/enterprise-guide/sso-keycloak.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+Let's say we want email, name, and surname to be passed between the client (Metabase) and the authentication server (Keycloak).
+
+1. Select “X500 email”, “X500 givenName” and “X500 surname” with the checkboxes that are on the right side of the screen.
+2. Click **Add Selected**.
+3. Click **Edit** beside each attribute and make the following changes:
+  - SAML Attribute Name: the name that Metabase expects to receive.
+  - SAML Attribute NameFormat: select “Basic” from the dropdown menu.
+
+Note: You can find the attribute values from **Admin settings** > **Authentication** > **<your SSO provider>** > **<Attributes>**.
```
Comment (jeff-bruemmer) [inline docs/enterprise-guide/sso-keycloak.md:53]:
```suggestion
You can find the attribute values from **Admin settings** > **Authentication** > **<your SSO provider>** > **<Attributes>**.
```

File: docs/enterprise-guide/sso-keycloak.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+
+## Check if SSO is working correctly
+Go to your Metabase login page. If SSO is working correctly, you should see a single button to sign-in with your IdP. Once you are authenticated, you should be automatically redirected to the Metabase home page.
```
Comment (jeff-bruemmer) [inline docs/enterprise-guide/sso-keycloak.md:57]:
```suggestion
Go to your Metabase login page. If SSO is working correctly, you should see a single button to sign in with your IdP. Once you're authenticated, you should be automatically redirected to the Metabase home page.
```

File: docs/enterprise-guide/sso-keycloak.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+  - Click **Set Password** to save your changes. 
+4. Create a new SSO client from **Manage** > **Clients** > **Create**.
+  - Client ID: Enter “metabase” in lowercase.
+  - Client Protocol: Select “saml” from the dropdown.
+  - Click **Save**.
+5. Configure the SSO client from the form that appears after saving:
+  - Client Signature Required: DISABLE
+  - Valid Redirect URIs: URL where you are hosting your Metabase instance followed by
+a slash (/) and an asterisk (*). For example, `http://localhost:3000/*`.
+  - Base URL: “URL the IdP should redirect back to” under Metabase **Admin settings** > **Authentication** > **<your SSO provider>**. For example, `“http://localhost:3000/auth/sso”`.
```
Comment (jeff-bruemmer) [inline docs/enterprise-guide/sso-keycloak.md:19]: This doesn't render well:

image

I'm guessing it's the `<`s screwing up the markdown renderer.

File: docs/enterprise-guide/sso-keycloak.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  - Click **Save**.
+6. Map user data to your SSO client from **Mappers** > **Add Builtin**.
+  - [Mapping attributes from users in Keycloak to Metabase](#mapping-attributes-from-users-in-keycloak-to-metabase)
+7. Configure the service provider (Metabase) from **Configure** > **Realm Settings**.
+  - Look for **Endpoints** and select “SAML 2.0 Identity Provider Metadata”. An XML file will open in a new tab.
+8. From the XML file, note the following:
```
Comment (jeff-bruemmer) [inline docs/enterprise-guide/sso-keycloak.md:25]: This also doesn't render well.

File: docs/enterprise-guide/sso-keycloak.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+  a. The URL that appears right after `md:SingleSignOnServiceBinding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location=`
+  b. The URL that appears right after `“entityID”`.
+  c. The long string that appears after the `<X509Certificate>` tag.
+
+
+## Mapping fields from Keycloak to Metabase
+
+1. Go to **Admin settings** > **Authentication** > **<your SSO provider>**.
```
Comment (jeff-bruemmer) [inline docs/enterprise-guide/sso-keycloak.md:33]: Another rendering issue.

**Review by jeff-bruemmer — CHANGES_REQUESTED** [review]
Great additions, @nllho.  Just a few notes, and some rendering issues to fix up before merging.

Also: be sure to add links to these new pages from both the main docs readme and the enterprise guide index page.

File: docs/enterprise-guide/sso-keycloak.md (hunk 2 above)
Comment (paoliniluis) [reply docs/enterprise-guide/sso-keycloak.md:1]: thanks!

File: docs/administration-guide/sso.md [hunk 13]
```diff
@@ -1,16 +1,34 @@
...
+- LDAP advanced features
+  - [Group membership filter][ldap-group-membership-filter]
+  - [Syncing user attributes][ldap-user-attributes]
+- [SAML][saml]
+  - [Setting up SAML with Auth0][saml-auth0]
+  - [Setting up SAML with Google][saml-google]
+  - [Setting up SAML with Keycloak][saml-keycloak]
+  - [Okta documentation](https://developer.okta.com/docs/guides/saml-application-setup/overview/)
+  - [OneLogin documentation](https://onelogin.service-now.com/support?id=kb_article&sys_id=83f71bc3db1e9f0024c780c74b961970)
+
```
Comment (nllho) [inline docs/administration-guide/sso.md:24]: @jeff-bruemmer 

I thought it would be helpful to include a list of common IdPs on the landing pages, but do you have any opinions about including dev docs here along with our Metabase guides?

File: docs/enterprise-guide/saml-keycloak.md [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+- **Client ID**: Enter “metabase” in lowercase.
+- **Client Protocol**: Select “saml” from the dropdown.
+- Click **Save**.
+
+5. Configure the SSO client from the form that appears after saving:
+
+- **Client Signature Required**: DISABLE
+- **Valid Redirect URIs**: URL where you are hosting your Metabase instance followed by
+  a slash (/) and an asterisk (_). For example, `http://localhost:3000/_`.
+- **Base URL**: Fill this in with the value under “URL the IdP should redirect back to” from your Metabase **Admin settings** > **Authentication** > **SAML**.
```
Comment (nllho) [inline docs/enterprise-guide/saml-keycloak.md:31]: oops, I just noticed this numbering is way off -- working on it 🤦‍♀️

File: docs/enterprise-guide/saml-keycloak.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+   - Turn off the **Temporary** toggle.
+   - Click **Set Password** to save your changes.
+4. Create a new SSO client from **Manage** > **Clients** > **Create**.
+   - **Client ID**: Enter “metabase” in lowercase.
+   - **Client Protocol**: Select “saml” from the dropdown.
+   - Click **Save**.
+5. Configure the SSO client from the form that appears after saving:
+   - **Client Signature Required**: DISABLE
+   - **Valid Redirect URIs**: URL where you are hosting your Metabase instance followed by
+     a slash (/) and an asterisk (*). For example, `http://localhost:3000/*`.
```
Comment (nllho) [inline docs/enterprise-guide/saml-keycloak.md:25]: This seems to render fine... but the highlighting is weird. Should I use the { raw } tag or something just to make sure?

@jeff-bruemmer let me know 🤔

File: docs/administration-guide/sso.md (hunk 13 above)
Comment (jeff-bruemmer) [reply docs/administration-guide/sso.md:24]: For cases where we don't want to (and shouldn't) take on documenting something ourselves, such as documenting third-party tooling, I'm for including links to helpful, relevant resources. We should, however, separate these resources into their own section with a heading that clearly marks them as external resources.

File: docs/enterprise-guide/saml-keycloak.md (hunk 14 above)
Comment (jeff-bruemmer) [reply docs/enterprise-guide/saml-keycloak.md]: Check out: https://github.com/metabase/metabase/pull/22930

File: docs/enterprise-guide/saml-keycloak.md (hunk 15 above)
Comment (jeff-bruemmer) [reply docs/enterprise-guide/saml-keycloak.md:25]: The highlighting looked okay to me, but I'm still seeing some numbering/nesting issues with the lists (e.g., in the Mapping Fields from Keycloak to Metabase section).

image

**Review by jeff-bruemmer — APPROVED** [review]
